server/src/main/java/org/opensearch/extensions/action/ExtensionActionUtil.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.action;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.io.stream.StreamInput;
+import java.lang.reflect.Constructor;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+/**
+ * ExtensionActionUtil - a class for creating and processing remote requests using byte arrays.
+ */
+public class ExtensionActionUtil {
+    private static final Logger logger = LogManager.getLogger(ExtensionActionUtil.class);
+
+    /**
+     * The Unicode UNIT SEPARATOR used to separate the Request class name and parameter bytes
+     */
+    public static final byte UNIT_SEPARATOR = (byte) '\u001F';
+
+    /**
+     * @param request r is an object of the "Remote Extension Action Request" class, containing information about the
+     * request being sent to the remote server. It is used to create a byte array containing the request data,
+     * which will be sent to the remote server.
+     * @return a byte array containing all the necessary information about the request to be sent to the remote server.
+     * This byte array is constructed using the class name of the request, a unit separator, and the request data itself.
+     */
+    public static byte[] createProxyRequestBytes(ExtensionActionRequest request) {
+        String requestClass = request.getClass().getName();
+        byte[] requestClassBytes = requestClass.getBytes(StandardCharsets.UTF_8);
+        return ByteBuffer.allocate(requestClassBytes.length + 1 + request.getRequestBytes().length)
```

Comment:
rather than request.getRequestBytes() here, you need to do the streamoutput collection of the request using writeTo() here.